File: Dummy/hawkshaw/library/ktextensions/StringsKt.java
package com.hawkshaw.library.ktextensions;

import Y1.K;
import java.util.Locale;
import r3.j;

public final class StringsKt {
    public static final String getEmpty(CharSequence charSequence) {
        return (charSequence == null || j.U(charSequence)) ? "" : charSequence.toString();
    }

    public static final int hash(String str) {
        K.n(str, "<this>");
        String lowerCase = str.toLowerCase(Locale.ROOT);
        K.m(lowerCase, "toLowerCase(...)");
        return Math.abs(lowerCase.hashCode());
    }

    public static final int hash(CharSequence charSequence) {
        K.n(charSequence, "<this>");
        return hash(charSequence.toString());
    }
}

File: app/src/main/java/com/hawkshaw/library/ktextensions/StringsKt.kt
package com.hawkshaw.library.ktextensions

import java.util.Locale
import kotlin.math.abs

/**
 * Utility functions for string operations
 */

/**
 * Get an empty string if the CharSequence is null or empty
 */
fun CharSequence?.getEmpty(): String {
    return if (this.isNullOrEmpty()) "" else this.toString()
}

/**
 * Calculate a non-negative hash code for a string
 */
fun String.hash(): Int {
    return abs(this.lowercase(Locale.ROOT).hashCode())
}

/**
 * Calculate a non-negative hash code for a CharSequence
 */
fun CharSequence.hash(): Int {
    return this.toString().hash()
} 